src/schema/show.ts

```diff
@@ -507,6 +507,25 @@ export const ShowType = new GraphQLObjectType({
         })
       },
     },
+    is_followed: {
+      type: GraphQLBoolean,
+      description: "Is the user following this show",
+      resolve: async (
+        show,
+        _args,
+        _context,
+        { rootValue: { followedShowsLoader } }
+      ) => {
+        if (!followedShowsLoader) return null
+
+        const response = await followedShowsLoader({ size: 100 })
```

Comment:
This isn't great, you're fetching up to 100 shows to then perform, on the client (Metaphysics), a lookup. Instead, the API endpoint that backs `followedShowsLoader` should accept a param that returns whether or not the show is followed. This also is going to be very inefficient if you wind up selecting `is_followed` for a bunch of different shows in your UI.

Instead, the Gravity endpoint should be modified to be accept something like: `/api/v1/follow_shows?shows[]=..` which would return a list of shows that are followed (from the given input).

Then, we can add a loader in Metaphysics that will efficiently use that endpoint. Otherwise this implementation and then selecting `is_followed` for a list of shows in a UI is going to be trouble.

Here's an example of what I'm referring to for genes: https://github.com/artsy/metaphysics/blob/b93fcc1f6c7fc79ee5902675146fa88ef9031886/src/lib/loaders/loaders_with_authentication/gravity.ts#L27-L33

Reply:
you use `followedArtistsLoader` to retrieve a list of followed artists. you use `followedArtistLoader` specifically to calculate your `is_followed` field.

Reply:
I'm not sure I understand the problem, could you elaborate? As I understand it, retrieving the list of shows _once_ per GraphQL query will cache it in the data loader layer (as long as the URL is the same, which specifying `shows[]=` would break). So if we're asking for 20 shows, and want the `is_following` field on each, then the `followedShowsLoader` would call Gravity once and then reuse that response for the 19 other uses, right?

Reply:
So, you would get caching working within one request. As this same user scrolls and a new paginated Relay request is being made (I'm assuming it's that type of setup), you'll be repeating this exact same call within that request. The change that goes to `shows[]=...` will still be efficient because the data loader will batch those together (change your example to be using genes and check out the `requests` extension).

```
{
  genes {
    name
    is_followed
  }
}
```

Another issue is the size of the response. You're getting the JSON of up to 100 shows back. That's a pretty large payload! We have all the patterns in place (both in Metaphysics and Gravity) to efficiently handle checking which, of a list of entities (aka shows), you follow. It efficiently batches those calls on the MP side, and returns a much more minimal amount of JSON from the Gravity side (you _only_ get back the shows you actually follow). So I'm basically just recommending to follow that pattern. It's a small update to the Gravity API endpoint to support that, and following this loader pattern in Metaphysics using `trackedEntityLoaderFactory`. The current implementation is sort of an anti-pattern, since we have everything we need and existing patterns for this exact use case, so I think we should stick with those.

Reply:
Okay, that makes sense. Thanks 👍 I'll follow up with those changes.

Reply:
Oh, one other piece of context is that these aren't Relay containers, so we'll have one (big) fetch per user. Not ideal, but the Local Discovery UI design has specific constraints that make it necessary. Let's still move ahead with Matt's suggestions.